src/Zoomba/DesktopLibrary.py

```diff
@@ -296,6 +285,16 @@ def open_application(self, remote_url, alias=None, window_name=None, splash_dela
             desired_caps["forceMjsonwp"] = True
         application = webdriver.Remote(str(remote_url), desired_caps)
         self._debug('Opened application with session id %s' % application.session_id)
```

Comment:
Ok so taking a look at this internally and it does work provided the splash screen shares the same AppId as the final application. Also if an application is brought up through something like Batch script then this will also be the case.

If we wrap these lines in a try/except we should get the result we want though:

'''
        try:
            application = webdriver.Remote(str(remote_url), desired_caps)
            self._debug('Opened application with session id %s' % application.session_id)
        except WebDriverException as e:
            if not window_name:
                zoomba.fail(str(e))
'''

Tested this locally and all seems well.